src/common/src/types/scalar_impl.rs

```diff
@@ -300,6 +300,40 @@ impl ScalarImpl {
         }
         for_all_scalar_variants! { impl_all_get_ident, self }
     }
+
+    pub(crate) fn data_type(&self) -> Result<DataType> {
+        let data_type = match self {
+            ScalarImpl::Int16(_) => DataType::Int16,
+            ScalarImpl::Int32(_) => DataType::Int32,
+            ScalarImpl::Int64(_) => DataType::Int64,
```

Comment:
IIRC `timestamp with time zone` is also stored as `ScalarImpl::Int64` but has `DataType::Timestampz`.

Reply:
Thanks for mention, I will try to fix in follow pr.

Reply:
Just bringing it up. So that we are aware of this tricky issue in type system and can see which design is better. I do need see an immediate way to "fix" - in the current design, mapping from `ScalarImpl` to `DataType` is not unique; alternatively, we may add a chrono wrapper for `timestampz`.

Reply:
Let me see, I found that acutally I don't need this function, I refactor this part https://github.com/singularity-data/risingwave/pull/2508.